### PR TITLE
refactor zk dynamic configuration listener

### PR DESCRIPTION
**Pull request by Mulavar**
**What this PR does**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Review by AlexStocks — APPROVED** [review]

File: remoting/zookeeper/listener.go [hunk 1]
```diff
@@ -81,6 +76,56 @@ func (l *ZkEventListener) ListenServiceNodeEvent(zkPath string, listener remotin
...
+				var content string
+				if event.Type == zk.EventNodeDeleted {
+					action = remoting.EventTypeDel
+				} else {
+					// 2. Try to get new configuration value of the zk node
+					// Notice: The order of step 1 and step 2 cannot be swapped, if you get value(with timestamp t1)
+					// before re-set the watcher(with timestamp t2), and some one change the data of the zk node after
+					// t2 but before t1, you may get the old value, and the new value will not trigger the event.
+					contentBytes, _, err := l.Client.Conn.Get(event.Path)
+					if err != nil {
```
Comment (sanxun0325) [inline remoting/zookeeper/listener.go:109]: If an error occurs, need continue execution?

Comment (Mulavar) [reply remoting/zookeeper/listener.go:109]: you're right, it's better do not continue the execution, I'll fix it.

**Comment by LaurenceLiZhixin** [comment]
@Mulavar Check Conflict and I would merge it.

**Review by LaurenceLiZhixin — APPROVED** [review]

**Review by sanxun0325 — APPROVED** [review]
LGTM

**Pull request merged**